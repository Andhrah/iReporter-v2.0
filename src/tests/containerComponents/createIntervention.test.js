import React from 'react';
import { mount, shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../store';
import ConnectedCreateIntervention, { CreateIntervention } from '../../components/container/CreateIntervention';

const props = {
  createIntervention: jest.fn(),
};

const eventDescribe = {
  target: {
    value: 'We will intervene shortly',
    name: 'describe'
  }
};

const event = {
  preventDefault: jest.fn(),
};

describe('CreateIntervention test suite', () => {
  it('renders correctly', () => {
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <ConnectedCreateIntervention {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('handles change correctly', () => {
    const wrapper = shallow(
      <CreateIntervention {...props} />
    );
    expect(wrapper.instance().state.interventionCreated).toEqual(false);
    const interventionDescription = wrapper.instance().handleChange(eventDescribe);
    expect(typeof (interventionDescription)).toEqual('object');
  });

  it('handles submmit correctly', async () => {
    const wrapper = shallow(
      <CreateIntervention {...props} />
    );
    expect(wrapper.instance().state.interventionCreated).toEqual(false);
    await wrapper.instance().handleSubmit(event);
    await expect(wrapper.instance().state.interventionCreated).toEqual(true);
  });
});
