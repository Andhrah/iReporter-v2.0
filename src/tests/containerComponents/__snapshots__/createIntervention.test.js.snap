// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateIntervention test suite renders correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <MemoryRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "vksgj0",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "vksgj0",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <ConnectFunction
        createIntervention={[MockFunction]}
      >
        <CreateIntervention
          createIntervention={[Function]}
          history={Array []}
        >
          <Navbar
            email={null}
            firstname={null}
            lastname={null}
            username={null}
          >
            <div
              className="wrapper"
            >
              <header>
                <div
                  className="branding"
                >
                  <Link
                    to="/"
                  >
                    <a
                      href="/"
                      onClick={[Function]}
                    >
                      <img
                        alt="iReporter logo"
                        src={Object {}}
                      />
                    </a>
                  </Link>
                </div>
                <nav
                  className="navbar"
                  id="topnav"
                >
                  <ul>
                    <li>
                      <Link
                        to="/redflags"
                      >
                        <a
                          href="/redflags"
                          onClick={[Function]}
                        >
                          Red-flags
                        </a>
                      </Link>
                    </li>
                    <li>
                      <Link
                        to="/interventions"
                      >
                        <a
                          href="/interventions"
                          onClick={[Function]}
                        >
                          Interventions
                        </a>
                      </Link>
                    </li>
                    <li>
                      <Link
                        to="/create-redflag"
                      >
                        <a
                          href="/create-redflag"
                          onClick={[Function]}
                        >
                          Report Corruption
                        </a>
                      </Link>
                    </li>
                    <li>
                      <Link
                        to="/create-intervention"
                      >
                        <a
                          href="/create-intervention"
                          onClick={[Function]}
                        >
                          Call for Intervention
                        </a>
                      </Link>
                    </li>
                    <li>
                      <Link
                        to="/signup"
                      >
                        <a
                          href="/signup"
                          onClick={[Function]}
                        >
                          Sign Up
                          <i
                            className="fa fa-user-plus"
                          />
                        </a>
                      </Link>
                    </li>
                    <li>
                      <Link
                        to="/login"
                      >
                        <a
                          href="/login"
                          onClick={[Function]}
                        >
                          Sign In
                           
                          <i
                            className="fa fa-user"
                          />
                        </a>
                      </Link>
                    </li>
                    <li>
                      <button
                        className="icon"
                        onClick={[Function]}
                        to="javascript:void(0);"
                        type="button"
                      >
                        <i
                          className="fa fa-bars"
                        />
                      </button>
                    </li>
                  </ul>
                </nav>
              </header>
            </div>
          </Navbar>
          <InterventionForm
            handleChange={[Function]}
            handleSubmit={[Function]}
            values={
              Object {
                "badHousing": "No electricity",
                "badroad": "Economy",
                "collapsedBridge": "Collapse",
                "economy": "Bad road",
                "interventionDate": [Function],
                "interventionImages": false,
                "interventionVideos": false,
                "noElectricity": "No electricity",
                "otherReasons": false,
                "otherReasonsSpecify": false,
                "poorElectricity": "Poor electricity",
                "refuseDamping": "No provision refuse damping",
              }
            }
          >
            <div
              className="report-form-container"
            >
              <h3>
                Call for Government
                 
                <span>
                  Intervention
                </span>
              </h3>
              <form
                className="report-form"
                onSubmit={[Function]}
              >
                <p>
                  Reasons for intervention
                </p>
                <div
                  className="first-checkbox-container"
                >
                  <div
                    className="checkbox-container"
                  >
                    <input
                      name="economy"
                      onChange={[Function]}
                      type="checkbox"
                      value="Bad road"
                    />
                    <label
                      htmlFor="economy"
                    >
                      Economy
                    </label>
                  </div>
                  <div
                    className="checkbox-container"
                  >
                    <input
                      name="badroad"
                      onChange={[Function]}
                      type="checkbox"
                      value="Economy"
                    />
                    <label
                      htmlFor="badroad"
                    >
                      Bad road
                    </label>
                  </div>
                  <div
                    className="checkbox-container"
                  >
                    <input
                      name="noElectricity"
                      onChange={[Function]}
                      type="checkbox"
                      value="No electricity"
                    />
                    <label
                      htmlFor="noElectricity"
                    >
                      No electricity
                    </label>
                  </div>
                  <div
                    className="checkbox-container"
                  >
                    <input
                      name="poorElectricity"
                      onChange={[Function]}
                      type="checkbox"
                      value="Poor electricity"
                    />
                    <label
                      htmlFor="poorElectricity"
                    >
                      Poor electricity
                    </label>
                  </div>
                  <div
                    className="checkbox-container"
                  >
                    <input
                      name="refuseDamping"
                      onChange={[Function]}
                      type="checkbox"
                      value="No provision refuse damping"
                    />
                    <label
                      htmlFor="refuseDamping"
                    >
                      No provision refuse damping
                    </label>
                  </div>
                  <div
                    className="checkbox-container"
                  >
                    <input
                      name="badHousing"
                      onChange={[Function]}
                      type="checkbox"
                      value="No electricity"
                    />
                    <label
                      htmlFor="badHousing"
                    >
                      Housing under power tension
                    </label>
                  </div>
                  <div
                    className="checkbox-container"
                  >
                    <input
                      name="collapsedBridge"
                      onChange={[Function]}
                      type="checkbox"
                      value="Collapse"
                    />
                    <label
                      htmlFor="collapsedBridge"
                    >
                      Collapse
                    </label>
                  </div>
                  <div
                    className="checkbox-container"
                  >
                    <input
                      name="otherReasons"
                      onChange={[Function]}
                      type="checkbox"
                      value={false}
                    />
                    <label
                      htmlFor="otherReasons"
                    >
                      Others
                    </label>
                  </div>
                </div>
                <div
                  className="checkbox-container"
                >
                  <label
                    htmlFor="otherReasonsSpecify"
                  >
                    If others, please specify
                  </label>
                  <input
                    name="otherReasonsSpecify"
                    onChange={[Function]}
                    type="text"
                    value={false}
                  />
                </div>
                <p>
                  Where did it happen?
                </p>
                <label
                  htmlFor="namesInvolved"
                >
                  Location
                </label>
                <input
                  id="names-involved"
                  name="namesInvolved"
                  onChange={[Function]}
                  required={true}
                  style={
                    Object {
                      "fontSize": 17,
                      "height": 40,
                      "paddingLeft": 15,
                      "paddingRight": 15,
                    }
                  }
                />
                <div
                  className="checkbox-container"
                >
                  <p>
                    Please describe what happened?
                  </p>
                  <textarea
                    cols={50}
                    name="describe"
                    onChange={[Function]}
                    placeholder="Enter text here"
                    rows={10}
                  />
                </div>
                <button
                  className="report-button"
                  type="submit"
                >
                  Call For Intervention
                </button>
              </form>
            </div>
          </InterventionForm>
          <t
            store={
              t {
                "_data": Object {
                  "classNames": "",
                  "message": "",
                  "status": "",
                  "timer": 0,
                },
                "_nextHandlerId": 1,
                "_watchers": Array [
                  Object {
                    "id": 0,
                    "trigger": [Function],
                  },
                ],
              }
            }
          >
            <Portal
              containerInfo={
                <body>
                  <div
                    class="toasts-container "
                    style="bottom: 10px; right: 10px;"
                  />
                </body>
              }
            >
              <div
                className="toasts-container "
                style={
                  Object {
                    "bottom": 10,
                    "right": 10,
                  }
                }
              />
            </Portal>
          </t>
        </CreateIntervention>
      </ConnectFunction>
    </Router>
  </MemoryRouter>
</Provider>
`;
